sdk/cosmosdb/cosmos/src/client/Container/Container.ts

```diff
@@ -318,15 +318,17 @@ export class Container {
    * @returns all the feed ranges for which changefeed could be fetched.
    */
   public async getFeedRanges(): Promise<ReadonlyArray<FeedRange>> {
-    const { resources } = await this.readPartitionKeyRanges().fetchAll();
+    return withDiagnostics(async (diagnosticNode: DiagnosticNodeInternal) => {
+      const { resources } = await this.readPartitionKeyRanges().fetchAllInternal(diagnosticNode);
 
-    const feedRanges: FeedRange[] = [];
-    for (const resource of resources) {
-      const feedRange = new FeedRangeInternal(resource.minInclusive, resource.maxExclusive);
-      Object.freeze(feedRange);
-      feedRanges.push(feedRange);
-    }
-    return Object.freeze(feedRanges);
+      const feedRanges: FeedRange[] = [];
+      for (const resource of resources) {
+        const feedRange = new FeedRangeInternal(resource.minInclusive, resource.maxExclusive);
+        Object.freeze(feedRange);
+        feedRanges.push(feedRange);
+      }
+      return feedRanges;
```

Comment:
I think complete array needs to be frozen for allowing no modifications in JS.

Reply:
So, the `withDiagnostics` needs to modify the array to inject diagnostic object to it. Also we don't freeze other objects, so I think we can avoid freezing this array.

Reply:
With this change it is possible to remove/update a feedRange from this list.